### PR TITLE
update workflow base OS and container to ubuntu:noble.

### PR DESCRIPTION
**Pull request by fujitatomoya**
follow up https://github.com/ros2/rclc/pull/407, see https://github.com/ros2/rclc/pull/407#issuecomment-2379968586

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,10 +15,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-22.04 ]
+        os: [ ubuntu-24.04 ]
```
Comment (fujitatomoya) [inline .github/workflows/ci.yml:18]: This does not really affect the workflow since CI is running in docker container but updated accordingly.

**Comment by fujitatomoya** [comment]
@JanStaschulat @pablogs9 @christophebedard either of you can take a look?

**Review by pablogs9 — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@JanStaschulat @pablogs9 can you merge this?

**Review by christophebedard — APPROVED** [review]
Looks good to me.

**Comment by fujitatomoya** [comment]
@JanStaschulat can you merge this? i do not think this does any harm.

**Review by JanStaschulat — APPROVED** [review]

**Pull request merged**